### PR TITLE
Ct 3915 setup new sar ir case type

### PR DESCRIPTION
**Pull request by willmcb**
## Description
It adds:
- `Case::SAR::InternalReview` inherited from the standard Sar
- Workflow for internal review
- Tests for new model
- Policy object + placeholder test file for this if there is more to add to it.
- Data migration for new correspondence type
- Some factories (Note commented out code here will likely be useful in further iterations so I have kept it in for now)

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [ ] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [ ] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### Related JIRA tickets
[CT-3927](https://dsdmoj.atlassian.net/browse/CT-3927)

### CODE REVIEW

File: config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+                add_message_to_case:
+                  if: Workflows::Predicates#user_is_approver_on_case?
+                reassign_user:
+                  if: Workflows::Predicates#case_is_assigned_to_responder_or_approver_in_same_team_as_current_user
+                  after_transition: Workflows::Hooks#reassign_user_email
+                unaccept_approver_assignment:
+                  if: Case::BasePolicy#can_unaccept_approval_assignment?
+                unflag_for_clearance:
+                  if: Workflows::Predicates#case_can_be_unflagged_for_clearance?
+                  switch_workflow: standard
```
Comment (ymao2) [inline config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml:186]: I can see there is a button to switch the trigger-flow back to standard,  based on my understanding, we shouldn't have such switching function, but anyway you can check with Nicole 


Actually,  there is something else coming to my mind, that there is another option is that no need define a duplicated workflow of SAR existing flow, just point to the same flow,   by using `def self.state_machine_name`

**Review by ymao2 — COMMENTED** [review]
You may need to add the new case type into 00_preamble.yml file too. 
Nice job 👍 , couple of  comments,  happy to discuss them

File: app/models/case/sar/internal_review.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Case::SAR::InternalReview < Case::SAR::Standard
+  class << self
+    def type_abbreviation
+      # This string is used when constructing paths or methods in other parts of
+      # the system. Ensure that it does not come from a user-supplied parameter,
+      # and does not contain special chars like slashes, etc.
+      'SAR_INTERNAL_REVIEW'
+    end
+  end
+end
```
[the remark this answers is not in the document]
Comment (willmcb) [reply app/models/case/sar/internal_review.rb:10]: Yep good idea will add that in!

File: config/settings.yml [hunk 3]
```diff
@@ -299,3 +299,10 @@ enabled_features:
     Host-staging: false
     Host-qa: false
     Host-prod: false
+  sar_internal_review:
+    Local: true
+    Host-dev: true
+    Host-demo: false
+    Host-staging: false
+    Host-qa: false
+    Host-prod: false
```
[the remark this answers is not in the document]
Comment (willmcb) [reply config/settings.yml:308]: Yep I saw that - thought I'd just add it in here as its sort of a form of config.

File: config/state_machine/configs/00_moj/60_sar_internal_review/10_sar_internal_review_preamble.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#########################################################################
+#                                                                       #
+#                         SAR INTERNAL REVIEW                           #
+#                                                                       #
+#########################################################################
+  sar_internal_review:
+    name: Subject Access Request Internal Review
+    permitted_workflows:
+      - standard
```
[the remark this answers is not in the document]
Comment (willmcb) [reply config/state_machine/configs/00_moj/60_sar_internal_review/10_sar_internal_review_preamble.yml:9]: Good spot thank you 👍

File: config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml (hunk 1 above)
Comment (willmcb) [reply config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml:186]: 🤔 that's a good idea I'll look into that.

File: config/state_machine/configs/00_moj/60_sar_internal_review/20_sar_internal_review_standard_workflow.yml [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+                flag_for_clearance:
+                  switch_workflow: trigger
+
+              drafting:
+                extend_sar_deadline:
+                  if: Workflows::Predicates#user_is_in_approving_team_for_case?
+                flag_for_clearance:
+                  switch_workflow: trigger
+                remove_sar_deadline_extension:
+                  if: Workflows::Predicates#case_extended_and_user_in_approving_team?
```
[the remark this answers is not in the document]
Comment (willmcb) [reply config/state_machine/configs/00_moj/60_sar_internal_review/20_sar_internal_review_standard_workflow.yml:138]: I checked and this is the case.

File: config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml (hunk 1 above)
Comment (willmcb) [reply config/state_machine/configs/00_moj/60_sar_internal_review/30_sar_internal_review_trigger_workflow.yml]: I have stuck with the original workflow and amended the predicate 👍

**Review by ymao2 — APPROVED** [review]

**Pull request merged**